packages/wdio-config/tests/FileSystemPathService.test.ts

```diff
@@ -28,66 +37,115 @@ describe('FileSystemPathService', () => {
         })
 
         it('should return current working directory', function () {
-            var svc = new FileSystemPathService()
+            const svc = new FileSystemPathService()
             expect(svc.getcwd()).toEqual(process.cwd())
         })
 
         it('should throw if cwd returns undefined', function () {
             process.cwd = () => undefined as any as string
-            var svc = new FileSystemPathService()
+            const svc = new FileSystemPathService()
             expect(() => svc.getcwd()).toThrowError('Unable to find current working directory from process')
         })
     })
 
     describe('isFile', function () {
-        it('should return true if file esxists', function () {
-            var svc = new FileSystemPathService()
+        it('should return true if file exists', function () {
+            vi.mocked(fs.existsSync).mockReturnValue(true)
+            vi.mocked(fs.lstatSync).mockReturnValue({
+                dev: 2114,
+                ino: 48064969,
+                mode: 33188,
+                nlink: 1,
+                uid: 85,
+                gid: 100,
+                rdev: 0,
+                size: 527,
+                blksize: 4096,
+                blocks: 8,
+                atimeMs: 1318289051000.1,
+                mtimeMs: 1318289051000.1,
+                ctimeMs: 1318289051000.1,
+                birthtimeMs: 1318289051000.1,
+                atime: new Date('Mon, 10 Oct 2011 23: 24: 11 GMT'),
+                mtime: new Date('Mon, 10 Oct 2011 23: 24: 11 GMT'),
+                ctime: new Date('Mon, 10 Oct 2011 23: 24: 11 GMT'),
+                birthtime: new Date('Mon, 10 Oct 2011 23: 24: 11 GMT'),
+                isFile: function (): boolean {
+                    return true
+                },
+                isDirectory: function (): boolean {
+                    return false
+                },
+                isBlockDevice: function (): boolean {
+                    return false
+                },
+                isCharacterDevice: function (): boolean {
+                    return false
+                },
+                isSymbolicLink: function (): boolean {
+                    return false
+                },
+                isFIFO: function (): boolean {
+                    return false
+                },
+                isSocket: function (): boolean {
+                    return false
+                }
```

Comment:
@christian-bromann: is there a way to do this?

Reply:
there is no need to create a mock for the complete object, just check what this specific function is using, e.g. in this case:

```js
return (fs.existsSync(filepath) && fs.lstatSync(filepath).isFile())
```
so you only need `isFile`